genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java

```diff
@@ -0,0 +1,279 @@
+import org.springframework.core.env.Environment;
+import org.springframework.core.task.AsyncTaskExecutor;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Component;
+
+import javax.script.Bindings;
+import javax.script.Compilable;
+import javax.script.CompiledScript;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+import javax.script.SimpleBindings;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * An implementation of the ClusterLoadBalancer interface which uses user a supplied script to make decisions based
+ * on the list of clusters and the job request supplied.
+ *
+ * The contract between the script and the Java code is that the script will be supplied global variables
+ * {@code clusters} and {@code jobRequest} which will be JSON strings representing the list (array) of clusters
+ * matching the cluster criteria tags and the job request that kicked off this evaluation. The code expects the script
+ * to either return the id of the cluster if one is selected or null if none was selected.
+ *
+ * @author tgianos
+ * @since 3.1.0
+ */
+@Component
+@ConditionalOnProperty(value = "genie.jobs.clusters.loadBalancers.script.enabled", havingValue = "true")
+@Slf4j
+public class ScriptLoadBalancer implements ClusterLoadBalancer {
+
+    static final String SCRIPT_TIMEOUT_PROPERTY_KEY = "genie.jobs.clusters.loadBalancers.script.timeout";
+    static final String SCRIPT_FILE_SOURCE_PROPERTY_KEY
+        = "genie.jobs.clusters.loadBalancers.script.source";
+    static final String SCRIPT_FILE_DESTINATION_PROPERTY_KEY
+        = "genie.jobs.clusters.loadBalancers.script.destination";
+    static final String SCRIPT_REFRESH_RATE_PROPERTY_KEY
+        = "genie.jobs.clusters.loadBalancers.script.refreshRate";
+    static final String SELECT_TIMER_NAME = "genie.jobs.clusters.loadBalancers.script.select.timer";
+    static final String UPDATE_TIMER_NAME = "genie.jobs.clusters.loadBalancers.script.update.timer";
+    static final String STATUS_TAG_KEY = "status";
+    static final String STATUS_TAG_OK = "ok";
+    static final String STATUS_TAG_NOT_FOUND = "not found";
+    static final String STATUS_TAG_NOT_CONFIGURED = "not configured";
+    static final String STATUS_TAG_FOUND = "found";
+    static final String STATUS_TAG_FAILED = "failed";
+    static final String EXCEPTION_TAG_KEY = "exception";
+
+    private static final long DEFAULT_TIMEOUT_LENGTH = 5_000L;
+    private static final Charset UTF_8 = Charset.forName("UTF-8");
+    private static final String SLASH = "/";
+    private static final String PERIOD = ".";
+    private static final int ORDER = Ordered.LOWEST_PRECEDENCE - 1;
```

Comment:
I agree with this. Lets do it as another feature in a different PR rather than bundling it in this one in the interest of getting this out.